src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocLeadingAsteriskAlignCheck.java

```diff
@@ -0,0 +1,240 @@
+        final DetailAST javadocEndToken = getBlockCommentAst().getLastChild();
+        final String lastLine = fileLines[javadocEndToken.getLineNo() - 1];
+        final OptionalInt asteriskColumnNumber =
+                    getAsteriskColumnNumberWithTabsExpanded(lastLine);
+
+        asteriskColumnNumber.ifPresent(integer -> {
+            checkAsteriskAlignment(javadocEndToken.getLineNo(),
+                    expectedColumnNumber,
+                    integer);
+        });
+    }
+
+    /**
+     * Processes and returns an Optional containing
+     * the column number of leading asterisk with tabs expanded.
+     * Also sets 'singleSpaceAsteriskColNumber' if the leading asterisk is present.
+     *
+     * @param line single line from block comment text
+     * @return column number of leading asterisk with tabs expanded
+     */
+    private OptionalInt getAsteriskColumnNumberWithTabsExpanded(String line) {
+        OptionalInt columnNumber = getAsteriskColumnNumber(line);
+
+        if (columnNumber.isPresent()) {
+            singleSpaceAsteriskColNumber = columnNumber.orElseThrow();
+            columnNumber = OptionalInt.of(CommonUtil.lengthExpandedTabs(
+                        line, columnNumber.orElseThrow(), getTabWidth()));
+        }
+
+        return columnNumber;
+    }
+
+    /**
+     * Processes and returns an Optional containing
+     * the column number of leading asterisk without tabs expanded.
+     *
+     * @param line javadoc comment line
+     * @return asterisk's column number
+     */
+    private static OptionalInt getAsteriskColumnNumber(String line) {
+        final Pattern pattern = Pattern.compile("^(\\s*)\\*");
+        final Matcher matcher = pattern.matcher(line);
+        final OptionalInt columnNumber;
+
+        // We may not always have a leading asterisk because a javadoc line can start with
+        // a non-whitespace character or the javadoc line can be empty.
+        // In such cases, there is no leading asterisk and Optional will be empty.
+        if (matcher.find()) {
+            columnNumber = OptionalInt.of(matcher.group(1).length() + 1);
+        }
+        else {
+            columnNumber = OptionalInt.empty();
+        }
+
+        return columnNumber;
+    }
+
+    /**
+     * Checks alignment of asterisks and logs violations.
+     *
+     * @param lineNumber line number of current comment line
+     * @param expectedColNumber column number of javadoc starting token
+     * @param asteriskColNumber column number of leading asterisk
+     */
+    private void checkAsteriskAlignment(int lineNumber,
+                                        int expectedColNumber,
+                                        int asteriskColNumber) {
+
+        final boolean hasValidAlignment = checkColumnDifference(expectedColNumber,
+                                                                asteriskColNumber);
+
+        if (!hasValidAlignment) {
```

Comment:
Please remove this as I've requested, we should not even enter this method if we don't need to.

Reply:
done. now it only logs violation:

https://github.com/checkstyle/checkstyle/blob/77dd1d104385ad79133495efee9f504f23f06730/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocLeadingAsteriskAlignCheck.java#L223-L232